main.tf

```diff
@@ -86,6 +86,7 @@ locals {
     ec2_ssh_key        = local.have_ssh_key ? var.ec2_ssh_key : "none"
     # Keep sorted so that change in order does not trigger replacement via random_pet
     source_security_group_ids = local.ng_needs_remote_access ? sort(var.source_security_group_ids) : []
+    additional_security_group_ids = sort(var.additional_security_group_ids)
```

Comment:
The `ng` object is just for data that is needed for the `aws_eks_node_group` resource. Since `additional_security_group_ids` only affects the launch template, this should not be here.